src/home_robot_hw/home_robot_hw/nodes/camera_pose_publisher.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import rospy
+import tf
+from geometry_msgs.msg import PoseStamped
+from home_robot.agent.motion.stretch import STRETCH_CAMERA_FRAME, STRETCH_BASE_FRAME
+from home_robot.utils.pose import to_matrix
+from home_robot_hw.ros.utils import matrix_to_pose_msg
+
+
+class CameraPosePublisher(object):
+    """ publishes the camera pose constantly so that we do not have a dependency on /tf"""
+
+    def __init__(self, topic_name="camera_pose"):
+        self._pub = rospy.Publisher(topic_name, PoseStamped)
+        self._listener = tf.TransformListener()
+        self._seq = 0
+
+    def spin(self, rate=10):
+        rate = rospy.Rate(rate)
+        while not rospy.is_shutdown():
+            try:
+                (trans, rot) = self._listener.lookupTransform("map", STRETCH_CAMERA_FRAME, rospy.Time(0))
+                matrix = to_matrix(trans, rot)
+                msg = PoseStamped(pose=matrix_to_pose_msg(matrix))
+                msg.header.stamp = rospy.Time.now()
```

Comment:
Just to confirm, the TransformListener gets triggered whenever an update is published to the tree, correct?
If so, is there a way to get the timestamp of the latest update since there might be a slight delay from the sensor data coming in to getting to this line where we assign the current time to the camera pose?

Reply:
I'm going to rewrite all this code to make sure it is a unified lookup between (rgb, depth, pose). I think that's the only way.

Reply:
yeah there is a way to get the timestamp of the last update though. maybe i'll add as an issue to make it clear i'm wokring on this